### ISSUE TITLE
bug(quorum-all-in-one): error "Failed to enumerate USB devices"

### ISSUE THREAD

**Issue by takeutak** [hyperledger/cacti#2274]
**Describe the bug**
In the process of the container building of quorum-all-in-one, there are the following error and the container cannot be launched.

```
[process]
DOCKER_BUILDKIT=1 docker build ./tools/docker/quorum-all-in-one/ -t qaio
docker run --rm qaio

[error]
"Failed to enumerate USB devices"
```

**To Reproduce**

When I edited [start-quorum.sh](https://github.com/hyperledger/cactus/blob/main/tools/docker/quorum-all-in-one/start-quorum.sh) and add "--nousb", the error does not happen.

```
geth \
  --identity node${NODE_ID}-${QUORUM_CONSENSUS:-raft} \
  --datadir ${DDIR} \
  --permissioned \
  --nodiscover \
  --verbosity 2 \
  --networkid ${NETWORK_ID} \
  --rpc \
  --rpccorsdomain "*" \
  --rpcvhosts "*" \
  --rpcaddr 0.0.0.0 \
  --rpcport 8545 \
  --rpcapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,quorum,${QUORUM_CONSENSUS:-raft} \
  --port 21000 \
  --allow-insecure-unlock \
  --unlock 0 \
  --password /passwords.txt \
  --nousb
  ${QUORUM_GETH_ARGS:-} ${GETH_ARGS_raft}
```
c.f.: https://github.com/ethereum/go-ethereum/issues/20568

**Comment by aldousalvarez** [comment]
Hello @petermetz I would like to work on this one, Thank you.

**Comment by ruzell22** [comment]
Hello @petermetz , can you assign this to me? Thank you.

**Comment by petermetz** [comment]
@aldousalvarez Sorry for the slow response!
@ruzell22 You got it.

**Comment by ruzell22** [comment]
Hi @takeutak , I have conducted multiple tests regarding the ticket including the one in the provided link.

Initial test without changes
![image](https://user-images.githubusercontent.com/89818529/229980360-355d18e2-1560-43ca-9e50-6dfbf07e40a3.png)

Added --nousb to replicate
![image](https://user-images.githubusercontent.com/89818529/229980430-a0018b29-93a5-420f-b0bc-87ebd0fad791.png)

changed --rpc flags to --http as per this log wherein it says it will be removed soon
![image](https://user-images.githubusercontent.com/89818529/229980561-bcc02752-5252-419f-9e08-4e012f54f54c.png)

The test without --nousb shows multiple occurrence of "Failed to enumerate USB devices" error log while the test with --nousb shows only one occurrence of the error log.

goquorum docs website also stated about adding --nousb will help remove the error
![image](https://user-images.githubusercontent.com/89818529/229980828-bcd1dee6-ba61-452c-86a7-3d3174297314.png)

I also tried adding the libusb and libusb-dev to the dockerfile and build and run the test again
![image](https://user-images.githubusercontent.com/89818529/229981025-d6495dfe-bf6d-4737-9f5f-c1fb820c4eaf.png)
![image](https://user-images.githubusercontent.com/89818529/229981068-00f36da1-f3fb-49a4-ae1f-d266c2548326.png)

"Failed to enumerate USB devices" error still happen even after adding --nousb flag. --privileged is also used when using docker run command.

**Comment by ruzell22** [comment]
Hi @takeutak , I would also like to confirm if the one below the logs, the part where it starts with raft, is it the expected output when running the container? Thank you.
![image](https://user-images.githubusercontent.com/89818529/230058729-86e3438e-88d0-4530-8da1-0535187d93cf.png)

**Comment by ruzell22** [comment]
Hello @takeutak and @petermetz , upon further testing, adding --nousb flag to the two geth command in start-quorum.sh removes all the "Failed to enumerate USB devices" error.
![image](https://user-images.githubusercontent.com/89818529/234168878-d8837611-85fa-4e84-9efe-864f7590655e.png)

The issue is fixed. However after the merge and update of quorum-all-in-one docker file in #2240 , with the new version of quorum and tessera, the /tessera/tessera-app.jar is now not accessible and is also changed to a .tar file.

I have tried to revert the version of the tessera, version of the quorum, and both at the same time to see which is causing the problem. The quorum-all-in-one is running as expected with the old and latest version of quorum, but it is only running okay with the old tessera, and is having error with new tessera version. The error is mostly regarding the extraction and running of /tessera/tessera-app.jar to its tar file equivalent.
![image](https://user-images.githubusercontent.com/89818529/234169887-85553447-049f-4efb-b48d-1414ccd4fd9c.png)

**Comment by ruzell22** [comment]
In relation to #2411 , based on the output on the terminal after building and running, "Failed to enumerate USB devices" error is not showing up anymore even without putting the --nousb flag in geth.

![image](https://github.com/hyperledger/cacti/assets/89818529/f3fcecab-2f86-4ba6-919b-8ae7b00b1fa6)
![image](https://github.com/hyperledger/cacti/assets/89818529/b3dceae1-094f-492d-a6ef-17136db848c5)

This bug has been fixed together with the #2411 PR, @petermetz , Thank you.

**Comment by petermetz** [comment]
@ruzell22 Thank you very much! Closing this one!
